src/main/java/javax/jmdns/impl/util/ExecutorServiceFactory.java

```diff
@@ -0,0 +1,49 @@
+package javax.jmdns.impl.util;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Factory class for creating customized {@link ExecutorService} instances with named threads.
+ * This class provides methods to create single-thread, cached, and fixed-thread pool executors,
+ * all of which utilize a {@link NamedThreadFactory} for thread naming.
+ *
+ * <p>Each method returns a new instance of an {@link ExecutorService} that is backed by a specific
+ * thread pool implementation.
+ */
+public class ExecutorServiceFactory {
+
+    /**
+     * Creates a new single-threaded executor service where all tasks are executed sequentially
+     * in a single thread.
+     *
+     * @param name the name to be used for the thread in this executor
+     * @return a newly created single-threaded {@link ExecutorService}
+     */
+    public ExecutorService newSingleThreadExecutor(String name) {
```

Comment:
Are these required to be instance methods (because of this multi-release jar) or could it be changed to static method like in case of `Executors`? 
▶️ if _yes_ then you don't need to instantiate this factory whenever you need an `ExecutorService`, and it would follow the logic of `Executors`. wdyt?

Reply:
I wasn't sure i noticed in a lot of the examples I saw none of the classes being replaced were `static` so i errored on the side of caution but I can try it.  It makes logical sense to be static

Reply:
Looks like it works as static!